From f1f25d2da9e6a7c0b28a333684449cc7733dc388 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Mon, 19 May 2025 14:37:58 +0000
Subject: [PATCH] backends/drm: also guard DrmOutput::cursorLayer for nullptr
 pipeline

BUG: 504516


(cherry picked from commit ceb2415e5bc314bb325ddc1664e684844f1a7ff3)

Co-authored-by: Xaver Hugl <xaver.hugl@gmail.com>
---
 src/backends/drm/drm_output.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/backends/drm/drm_output.cpp b/src/backends/drm/drm_output.cpp
index a467d737901..151cb2d466f 100644
--- a/src/backends/drm/drm_output.cpp
+++ b/src/backends/drm/drm_output.cpp
@@ -568,6 +568,11 @@ DrmOutputLayer *DrmOutput::primaryLayer() const
 
 DrmOutputLayer *DrmOutput::cursorLayer() const
 {
+    if (!m_pipeline) {
+        // this can happen when the output gets hot-unplugged
+        // FIXME fix output lifetimes so that this doesn't happen anymore...
+        return nullptr;
+    }
     return m_pipeline->cursorLayer();
 }
 
-- 
GitLab

